src/main/java/pl/foodRecipe/domain/recipe/RecipeService.java

```diff
@@ -0,0 +1,63 @@
+package pl.foodRecipe.domain.recipe;
+
+import org.springframework.stereotype.Service;
+import pl.foodRecipe.domain.category.Category;
+import pl.foodRecipe.domain.category.CategoryRepository;
+import pl.foodRecipe.domain.recipe.dto.RecipeDto;
+import pl.foodRecipe.domain.recipe.dto.RecipeSaveDto;
+import pl.foodRecipe.storage.FileStorageService;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class RecipeService {
+
+    private final RecipeRepository recipeRepository;
+    private final CategoryRepository categoryRepository;
+    private final FileStorageService fileStorageService;
+
+    public RecipeService(RecipeRepository recipeRepository,
+                         CategoryRepository categoryRepository,
+                         FileStorageService fileStorageService) {
+        this.recipeRepository = recipeRepository;
+        this.categoryRepository = categoryRepository;
+        this.fileStorageService = fileStorageService;
+    }
+
+
+    public List<RecipeDto> findAllPromotedRecipes() {
+        return recipeRepository.findAllByPromotedIsTrue().stream()
+                .map(RecipeDtoMapper::map)
+                .toList();
+    }
+
+    public Optional<RecipeDto> findRecipeById(long id) {
+        return recipeRepository.findById(id).map(RecipeDtoMapper::map);
+    }
+
+    public List<RecipeDto> findRecipesByCategoryName(String category) {
```

Comment:
Użyłbym page'owania żeby wyświetlać jako liste powiedzmy 50 encji, bo jeśli będziesz miał ich tam 1000 to jest problemik :)